app/src/main/java/ninja/knowone/apod/CallingNasa.kt

```diff
@@ -1,24 +1,37 @@
 package ninja.knowone.apod
 
 import android.app.Activity
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
 import android.os.AsyncTask
 import android.widget.Toast
+import com.bumptech.glide.Glide
+import kotlinx.android.synthetic.main.fragment_apod_main.*
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import org.json.JSONObject
 import android.app.DatePickerDialog.OnDateSetListener as OnDateSetListener1
 
 class CallingNasa(private val activity: Activity) {
 
+    private fun hasNetwork(): Boolean {
+        val connectivityManager = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val activeNetwork: NetworkInfo? = connectivityManager.activeNetworkInfo
+        return activeNetwork?.isConnectedOrConnecting == true
+    }
+
+
      fun picSnag(date:String = "", filePasser: (JSONObject) -> Unit) {
-        val client = OkHttpClient()
-        var url = "https://api.nasa.gov/planetary/apod?api_key=${activity.getString(R.string.api_key)}&hd=true"
-         if (!date.isBlank()) {
+         if(hasNetwork() == true) {
```

Comment:
Same here in an if, you can just give it `hasNetwork()` without comparing it to true.

Reply:
I wondered why Alt enter remove the Boolean. I thought it was wrong so I put it back 🤣